zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +150,238 @@ def compute(self, today, assets, out, close, volume):
+        - r_value, the correlation coefficient
+        - p_value, the two-sided p-value for a hypothesis test whose null
+          hypothesis is that the slope is zero.
+        - stderr, the standard error of the estimate
+
+    Example
+    -------
+    Let the following be example 10-day returns for three different assets::
+
+                       SPY    MSFT     FB
+        2017-03-13    -.03     .03    .04
+        2017-03-14    -.02    -.03    .02
+        2017-03-15    -.01     .02    .01
+        2017-03-16       0    -.02    .01
+        2017-03-17     .01     .04   -.01
+        2017-03-20     .02    -.03   -.02
+        2017-03-21     .03     .01   -.02
+        2017-03-22     .04    -.02   -.02
+
+    Suppose we are interested in predicting each stock's returns from SPY's
+    over rolling 5-day look back windows. We can compute rolling regression
+    coefficients (alpha and beta) from 2017-03-17 to 2017-03-22 by doing::
+
+        regression_factor = RollingRegressionOfReturns(
+            asset=Equity(24),
+            returns_length=10,
+            regression_length=5,
+        )
+        alpha = regression_factor.alpha
+        beta = regression_factor.beta
+
+    The result of computing ``alpha`` from 2017-03-17 to 2017-03-22 gives::
+
+                       SPY    MSFT     FB
+        2017-03-17       0    .011   .003
+        2017-03-20       0   -.004   .004
+        2017-03-21       0    .007   .006
+        2017-03-22       0    .002   .008
+
+    And the result of computing ``beta`` from 2017-03-17 to 2017-03-22 gives::
+
+                       SPY    MSFT     FB
+        2017-03-17       1      .3   -1.1
+        2017-03-20       1      .2     -1
+        2017-03-21       1     -.3     -1
+        2017-03-22       1     -.3    -.9
+
+    Note that SPY's column for alpha is all 0's and for beta is all 1's, as the
+    regression line of SPY with itself is simply the function y = x.
+
+    To understand how each of the other values were calculated, take for
+    example MSFT's ``alpha`` and ``beta`` values on 2017-03-17 (.011 and .3,
+    respectively). These values are the result of running a linear regression
+    predicting MSFT's returns from SPY's returns, using values starting at
+    2017-03-17 and looking back 5 days. That is, the regression was run with
+    x = [-.03, -.02, -.01, 0, .01] and y = [.03, -.03, .02, -.02, .04], and it
+    produced a slope of .3 and an intercept of .011.
+
+    See Also
+    --------
+    :class:`zipline.pipeline.factors.technical.RollingPearsonOfReturns`
+    :class:`zipline.pipeline.factors.technical.RollingSpearmanOfReturns`
+    """
+    outputs = ['alpha', 'beta', 'r_value', 'p_value', 'stderr']
+    params = ['baseline_asset']
+
+    def __new__(cls,
+                baseline_asset,
+                returns_length,
+                regression_length,
+                mask=NotSpecified,
+                **kwargs):
+        if mask is not NotSpecified:
```

Comment:
Do you mean that this `|` needs to be applied in the case of a class-level defined mask? I'm not sure how this could be applied unconditionally, i.e. if no mask is passed